tests/ledger_updates.rs

```diff
@@ -0,0 +1,243 @@
+        let db = setup_db("test-ledger-updates-by-milestone").await.unwrap();
+        let collection = setup_coll::<LedgerUpdateCollection>(&db).await.unwrap();
+
+        let ctx = bee_block_stardust::protocol::protocol_parameters();
+
+        let mut outputs = HashMap::new();
+        let ledger_outputs = std::iter::repeat_with(|| (BlockId::rand(), Output::rand_basic(&ctx), OutputId::rand()))
+            .take(100)
+            .enumerate()
+            .inspect(|(i, (_, _, output_id))| {
+                assert!(outputs.insert(*output_id, *i as u32 / 5).is_none());
+            })
+            .map(|(i, (block_id, output, output_id))| LedgerOutput {
+                block_id,
+                booked: MilestoneIndexTimestamp {
+                    milestone_index: (i as u32 / 5).into(),
+                    milestone_timestamp: (12345 + (i as u32 / 5)).into(),
+                },
+                output,
+                output_id,
+                rent_structure: RentStructureBytes {
+                    num_key_bytes: 0,
+                    num_data_bytes: 100,
+                },
+            })
+            .collect::<Vec<_>>();
+
+        assert_eq!(ledger_outputs.len(), 100);
+
+        collection
+            .insert_unspent_ledger_updates(ledger_outputs.iter())
+            .await
+            .unwrap();
+
+        assert_eq!(collection.count().await.unwrap(), 100);
+
+        let mut s = collection
+            .get_ledger_updates_by_milestone(0.into(), 100, None)
+            .await
+            .unwrap();
+
+        while let Some(LedgerUpdateByMilestoneRecord {
+            output_id, is_spent, ..
+        }) = s.try_next().await.unwrap()
+        {
+            assert_eq!(outputs.remove(&output_id), Some(0));
+            assert!(!is_spent);
+        }
+        assert_eq!(outputs.len(), 95);
+
+        teardown(db).await;
+    }
+
+    #[tokio::test]
+    async fn test_spent_unspent_ledger_updates() {
+        let db = setup_db("test-spent-unspent-ledger-updates").await.unwrap();
+        let collection = setup_coll::<LedgerUpdateCollection>(&db).await.unwrap();
+
+        let ctx = bee_block_stardust::protocol::protocol_parameters();
+
+        let mut booked_outputs = Vec::new();
+
+        let spent_outputs = std::iter::repeat_with(|| (BlockId::rand(), Output::rand_basic(&ctx), OutputId::rand()))
+            .take(100)
+            .map(|(block_id, output, output_id)| LedgerOutput {
+                block_id,
+                booked: MilestoneIndexTimestamp {
+                    milestone_index: 0.into(),
+                    milestone_timestamp: 10000.into(),
+                },
+                output,
+                output_id,
+                rent_structure: RentStructureBytes {
+                    num_key_bytes: 0,
+                    num_data_bytes: 100,
+                },
+            })
+            .inspect(|booked_output| {
+                assert!(!booked_outputs.contains(booked_output));
```

Comment:
So there are two options here, either this can realistically fail or it cannot. If it can, then we should be using login to ensure that they are unique. If not, then remove this check.